### PR TITLE
Fix reading of stale information on geoptions re-fetch

### PR DESCRIPTION
**Pull request by AdityaKhatri**

### CODE REVIEW

File: src/views/Entries/index.js [hunk 1]
```diff
@@ -145,20 +149,163 @@ const leadKeySelector = d => d.id;
...
+
+            return ({
+                filters: [
+                    ...processedFilters,
+                    ...Object.entries(otherFilters),
+                ],
+            });
+        },
+        onMount: false,
+        onPropsChanged: ['entriesFilter', 'activePage'],
```
Comment (tnagorra) [inline src/views/Entries/index.js:249]: Shouldn't this also be called on onMount?

Comment (AdityaKhatri) [reply src/views/Entries/index.js:249]: No, we only fetch entries after geooptions and frameworks are loaded to avoid unintended data references.

Comment (tnagorra) [reply src/views/Entries/index.js:249]: There are various limitations with this apporach when we have entriesFilter/activePage as triggers.
And, the data loading is also slow.
We should rather focus on fixing the bad-data reference (we can just hide entries unless geo-options and frameworks are loaded)

File: src/views/Entries/index.js [hunk 2]
```diff
@@ -307,19 +398,34 @@ export default class Entries extends React.PureComponent {
...
+        this.setState({
+            successGeoOptions: true,
+        }, this.startEntriesRequest);
+    }
+
+    handleFrameworkGetSuccess = () => {
+        this.setState({
+            successFramework: true,
+        }, this.startEntriesRequest);
+    }
```
Comment (tnagorra) [inline src/views/Entries/index.js:427]: Both of them starts entries request?

Comment (tnagorra) [reply src/views/Entries/index.js:427]: This looks very easy to break. We should change this behavior as well.

**Pull request merged**